delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/jsonld/JsonLdSourceParser.scala

```diff
@@ -40,12 +40,13 @@ trait JsonLdSourceParser {
     * @return a tuple with the resulting @id iri and the decode value
     */
   def decode[A: JsonLdDecoder, R](project: Project, source: Json)(implicit
+      context: ContextValue,
       uuidF: UUIDF,
       rcr: RemoteContextResolution,
       rejectionMapper: Mapper[JsonLdRejection, R]
   ): IO[R, (Iri, A)] = {
     for {
-      (_, expanded) <- expandSource(project, source)
+      (_, expanded) <- expandSource(project, source.addContext(context))
```

Comment:
You removed this line from the `Impl` just to add it here. I don't find it beneficial in terms of readability since this is done through an implicit `ContextValue` that won't get picked up automatically but it will have to be imported in the implementation itself.

I find it more obfuscating than instructing

Reply:
We already add context to the SearchResults encoder, via an implicit value, I thought it was similar and good enough.

Reply:
It isn't great for `SearchResults` too. And that happens in the routes and not on the Impl